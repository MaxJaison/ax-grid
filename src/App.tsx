import { Button, Center, Heading } from '@chakra-ui/react';
import { Link } from 'react-router-dom';

const App = () => {
    return (
        <Center bg="#68ede7" h="100vh" color="black" alignItems="center">
            <Heading as="h2" size="2xl">
                Welcome to AX Grid
            </Heading>
            <Button data-testid="link" colorScheme="blue" ml="10">
                <Link to="/grid">Enter Platform</Link>
            </Button>
        </Center>
    );
};

export default App;
