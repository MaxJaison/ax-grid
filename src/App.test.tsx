import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import App from './App';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/vitest';

describe('App Component', () => {
    it('renders the App component', async () => {
        render(<App />, { wrapper: BrowserRouter });

        expect(screen.getByText(/Welcome to AX Grid/i)).toBeInTheDocument();
    });

    it('create App snapshot', async () => {
        const { asFragment } = render(<App />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
