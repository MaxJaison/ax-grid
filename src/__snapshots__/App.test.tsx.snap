// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > create App snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-192kh8p"
  >
    <h2
      class="chakra-heading css-0"
    >
      Welcome to AX Grid
    </h2>
    <button
      class="chakra-button css-1fb3vhs"
      data-testid="link"
      type="button"
    >
      <a
        href="/grid"
      >
        Enter Platform
      </a>
    </button>
  </div>
</DocumentFragment>
`;
